### PR TITLE
Software part improvements

### PR DESCRIPTION
**Pull request by mattkaem**
- Added the possibility to enable and disable http and mqtt adapters
- Made key and crt files only mandatory for creation and modification of the kubernetes secrets

### CODE REVIEW

File: terraform/modules/hono/variables.tf [hunk 1]
```diff
@@ -71,22 +91,62 @@ variable "device_communication_dns_name" {
 
 variable "api_tls_key" {
   type        = string
-  description = "Filename of the ingress tls/server Key File including file extension"
+  description = "Content of the ingress tls/server Key File"
```
Comment (gdimitropoulos-sotec) [inline terraform/modules/hono/variables.tf:94]: is api_tls_key the file path or the key?

File: terraform/software/outputs.tf [hunk 2]
```diff
@@ -1,4 +1,34 @@
...
+  sensitive = true
+}
+
+output "http_tls_key_in_storage" {
+  value = var.enable_http_adapter ? (var.http_tls_key == null ? var.http_tls_key_from_storage : var.http_tls_key) : null
+  sensitive = true
+}
+
+output "http_tls_crt_in_storage" {
+  value = var.enable_http_adapter ? (var.http_tls_crt == null ? var.http_tls_crt_from_storage : var.http_tls_crt) : null
```
Comment (gdimitropoulos-sotec) [inline terraform/software/outputs.tf:22]: should be null or false?

Comment (gdimitropoulos-sotec) [reply terraform/software/outputs.tf:22]: because enable_http_adapter is bool

File: terraform/modules/hono/variables.tf (hunk 1 above)
Comment (mattkaem) [reply terraform/modules/hono/variables.tf:94]: No, it is the content of the key file

File: terraform/modules/hono/main.tf [hunk 3]
```diff
@@ -72,35 +80,53 @@ resource "kubernetes_namespace" "hono" {
...
   data = {
-    "tls.crt" = var.api_tls_crt
-    "tls.key" = var.api_tls_key
+    "tls.crt" = var.api_tls_crt == null ? var.api_tls_crt_from_storage : var.api_tls_crt
+    "tls.key" = var.api_tls_key == null ? var.api_tls_key_from_storage : var.api_tls_key
   }
 }
 
 resource "kubernetes_secret" "esp-ssl" {
+  count = (var.api_tls_crt != null || var.api_tls_crt_from_storage != null) && (var.api_tls_key != null || var.api_tls_key_from_storage != null) != null ? 1 : 0
```
[the remark this answers is not in the document]
Comment (mattkaem) [reply terraform/modules/hono/main.tf]: good catch, I changed it accordingly

File: terraform/modules/hono/main.tf [hunk 4]
```diff
@@ -72,35 +80,53 @@ resource "kubernetes_namespace" "hono" {
...
   data = {
-    "server.crt" = var.api_tls_crt
-    "server.key" = var.api_tls_key
+    "server.crt" = var.api_tls_crt == null ? var.api_tls_crt_from_storage : var.api_tls_crt
+    "server.key" = var.api_tls_key == null ? var.api_tls_key_from_storage : var.api_tls_key
+  }
+}
+
+resource "kubernetes_secret" "http_secret" {
+  count = var.enable_http_adapter && (var.http_tls_crt != null || var.http_tls_crt_from_storage != null) && (var.http_tls_key != null || var.http_tls_key_from_storage != null) != null ? 1 : 0
```
[the remark this answers is not in the document]
Comment (mattkaem) [reply terraform/modules/hono/main.tf]: see above answer

**Comment by gdimitropoulos-sotec** [comment]
Approved

**Pull request merged**